dwave/cloud/coders.py

```diff
@@ -237,4 +242,60 @@ def decode_qp_numpy(msg, return_matrix=True):
         result['active_variables'] = result['active_variables'].tolist()
         result['solutions'] = result['solutions'].tolist()
 
+    # include problem type
+    if 'type' in msg:
+        result['problem_type'] = msg['type']
+
+    return result
+
+
+def encode_problem_as_bq(problem, compress=False):
+    """Encode the binary quadratic problem for submission in the `bq` data
+    format.
+
+    Args:
+        problem (:class:`~dimod.BinaryQuadraticModel`):
+            Binary quadratic model.
+
+    Returns:
+        encoded submission dictionary
+    """
+
+    import zlib
+    import json
+
+    serialized_bqm = problem.to_serializable(use_bytes=False)
+
+    if compress:
+        # note: compression scheme will change, and preferably move to transport layer
+        compressed_bqm = zlib.compress(bytes(json.dumps(serialized_bqm), "ascii"))
```

Comment:
oh my!!
you sure you wanna leave this in?
I agree with using the transport layer instead of this also

Reply:
I though you want to use application layer compression, at least during dev. Are you not? I can remove it if you're sure.